### ISSUE TITLE
Running SRA tools on cloud worker machine

### ISSUE THREAD

**Issue by OrielResearchCure** [ncbi/sra-tools#316]
Hello,

I am trying to generate a configuration file that I can use on worker node without calling vdb-config.

I followed the steps with the newer versions:
1. use a file that was generated by vdb-config --restore-defaults
2. cp the generated config to my local folder
3. this is my final config file:
/config/default = "false"
/libs/cloud/accept_gcp_charges = "true"
/repository/remote/main/CGI/resolver-cgi = "https://trace.ncbi.nlm.nih.gov/Traces/names/names.fcgi"
/repository/remote/protected/CGI/resolver-cgi = "https://trace.ncbi.nlm.nih.gov/Traces/names/names.fcgi"
/repository/user/ad/public/apps/file/volumes/flatAd = "/content/datalab/notebooks"
/repository/user/ad/public/apps/refseq/volumes/refseqAd = "/content/datalab/notebooks"
/repository/user/ad/public/apps/sra/volumes/sraAd = "/content/datalab/notebooks"
/repository/user/ad/public/apps/sraPileup/volumes/ad = "/content/datalab/notebooks"
/repository/user/ad/public/apps/sraRealign/volumes/ad = "/content/datalab/notebooks"
/repository/user/ad/public/root = "/content/datalab/notebooks"
/repository/user/default-path = "/content/datalab/notebooks"
/repository/user/main/public/root = "/content/datalab/notebooks"
/tools/prefetch/download_to_cache = "false"

4. export VDB_CONFIG=/content/datalab/notebooks/GEO_RNA_custom.kfg
5. vdb-config /repository/user/main => generate and output like that:
<main>
  <public>
    <apps>
      <file>
        <volumes>
          <flat>files</flat>
        </volumes>
      </file>....
6. Call (python code): subprocess.run(commands,shell=True) with and without export, gave the same error:
sra_toolkit_folder = "/content/datalab/notebooks/sratoolkit.2.10.5-centos_linux64/bin"
srr = 'SRR4784332'
commands = "export VDB_CONFIG=/content/datalab/notebooks/GEO_RNA_custom.kfg; {}/fasterq-dump -p {}".format(sra_toolkit_folder,srr)
print("commands are  {}".format(commands))
subprocess.check_output(commands,shell=True)

Error:
CalledProcessError: Command 'export VDB_CONFIG=/content/datalab/notebooks/GEO_RNA_custom.kfg; /content/datalab/notebooks/sratoolkit.2.10.5-centos_linux64/bin/fasterq-dump -p SRR4784332' returned non-zero exit status 78

Please advice what am I missing
Thanks,
eilalan

The main technique for doing this today is somewhat kludgy. First, let me say why.

The constraint is that prefetch is a user-level tool working on a personal user's cache. Bombarding a single user's cache from a cluster is generally not a good thing. We have a concept of a "site" repository that allows a single cache to be used from several users/places, but the repository admin still has to take care to manage potential collisions when populating the cache.

In your case, you seem to want to prefetch rather than using https to get data on demand - I presume you're using Aspera? If not, just use https.

For the multiple caches per single user, the way to do this today would be by creating a custom configuration file per job on your cluster. The first part would be to start with a copy of our default configuration file: see https://github.com/ncbi/ncbi-vdb/blob/master/libs/kfg/default.kfg . You'll want to edit the line 38 (as of 4/24/2017) that reads

`/repository/user/main/public/root` = "$(HOME)/ncbi/public"

and replace the path on the right hand side of the assignment operator. There are 2 very important rules to follow or it will not work:
1. the path MUST be in double quotes
2. do not use variable substitution - it doesn't work the way you might think

Replace it with something like

`/repository/user/main/public/root` = "/net/myvol/blah/public"

Assuming that you don't have anything else within your user settings to alter the public user cache and don't have a site-wide configuration, this file should be enough to provide for general toolkit configuration.

The means of getting your toolkit to recognize this file rather than the one it currently sees is by setting and exporting the shell variable `VDB_CONFIG`.

So in summary:

`cp /path/to/default.kfg /my-cluster-workspace/custom.kfg`  
update the custom.kfg using whatever editor you like - sed, awk, etc.  
`export VDB_CONFIG=/my-cluster-workspace/custom-kfg`  
at this point you should be good to go, and can test the path with `vdb-config`

`vdb-config /repository/user/main`

_Originally posted by @kwrodarmer in https://github.com/ncbi/sra-tools/issues/71#issuecomment-296830659_

**Comment by klymenko** [comment]
@OrielResearchCure, do you still need help?
